server/src/main/java/keywhiz/auth/cookie/AuthenticatedEncryptedCookieFactory.java

```diff
@@ -92,13 +88,18 @@ public String getSession(User user, ZonedDateTime expiration) {
    */
   public NewCookie cookieFor(String value, ZonedDateTime expiration) {
     long maxAge = Duration.between(ZonedDateTime.now(clock), expiration).getSeconds();
+    Date expirationDate = Date.from(expiration.toInstant());
 
-    HttpCookie cookie = new HttpCookie(config.getName(), value, config.getDomain(),
-        config.getPath(), maxAge, config.isHttpOnly(), config.isSecure());
+    // If the cookie is already expired, set it to have no lifespan (maxAge = 0) and
+    // an expiration in the distant past. This is particularly important because a maxAge
+    // value of -1 would mark that the cookie expired at the end of the current session.
```

Comment:
I wonder if that's some special handling in [JAX-RS](https://jax-rs.github.io/apidocs/2.1/), since I didn't see anything special about `Max-Age: -1` in either [RFC 2109](https://tools.ietf.org/html/rfc2109) or [RFC 6265](https://tools.ietf.org/html/rfc6265#section-5.2.2) 🤔